src/Sylius/Bundle/ResourceBundle/Controller/ParametersParser.php

```diff
@@ -77,6 +78,16 @@ private function parseRequestValue($parameter, Request $request)
             return $this->parseRequestValueExpression(substr($parameter, 5), $request);
         }
 
+        if (0 === strpos($parameter, '!!')) {
+            $parameter = explode(' ', $parameter);
+
+            $parser = trim($parameter[0], '!').'val';
+
+            Assert::oneOf($parser, ['intval', 'floatval', 'doubleval'], 'Variable can be parsed only to int, float or double.');
+
+            return $parser($request->get(substr($parameter[1], 1)));
```

Comment:
This would fail unless the third parameter of `explode` is set to `2` (`explode(' ', 'a b c') === ['a', 'b', 'c']` -> `explode(' ', 'a b c', 2) === ['a', 'b c']`). We need some tests for data with spaces in it.